python_modules/dagster/dagster_tests/definitions_tests/test_dynamic_partitions.py

```diff
@@ -204,3 +210,80 @@ def test_has_partition_key():
         assert partitions_def.has_partition_key("apple", dynamic_partitions_store=instance)
         assert partitions_def.has_partition_key("banana", dynamic_partitions_store=instance)
         assert not partitions_def.has_partition_key("peach", dynamic_partitions_store=instance)
+
+
+def test_dynamic_outputs_dynamic_partitions():
+    """Test the interoperability of dagster's dynamic output system with the dynamic partitioning system."""
+    partitions = ["apple", "banana", "cantaloupe"]
+    values_per_partition = {
+        "apple": 1,
+        "banana": 2,
+        "cantaloupe": 3,
+    }
+
+    partitions_def = DynamicPartitionsDefinition(name="fruits")
+
+    @dg.op(out=dg.DynamicOut())
+    def fan_out(context: dg.OpExecutionContext):
+        for partition_key in context.partition_keys:
+            yield dg.DynamicOutput(values_per_partition[partition_key], mapping_key=partition_key)
+
+    @dg.op
+    def process_fruit_val(context, val: int):
+        return val + 1
+
+    @dg.op(out=dg.Out(dagster_type=dg.Nothing))
+    def collect_fruit_vals(context, vals):
+        assert vals == [2, 3, 4]
+
+    @dg.graph_asset(partitions_def=partitions_def, backfill_policy=dg.BackfillPolicy.single_run())
+    def doubly_dynamic_asset():
+        vals = fan_out().map(process_fruit_val)
+        return collect_fruit_vals(vals.collect())
+
+    with instance_for_test() as instance:
+        instance.add_dynamic_partitions(dg._check.not_none(partitions_def.name), partitions)  # noqa: SLF001
```

Comment:
In this case we are adding the dynamic partitions before executing anything. In a real deployment this would happen where, in a sensor that then requests this run? 

Is there a similar but slightly different example that we would want to have that calculates and adds the dynamic partitions as part of the run? Is that possible?

Reply:
In a real deployment it would happen in a sensor. I see the utility of having a run that calculates and adds the dynamic partitions as well as processing, but it is not currently possible. For a run to be partitioned (that is, for all the partitioned properties to be filled out), you need to know the partition keys ahead of time (at least right now)